File: scripts/dummyContract.py
from web3 import Web3

contractAddress = '0x0103CfdA95f2D6084d28a935db26Ff1C5cd7DC6d'
contractAbi = [
    {
      "inputs": [],
      "stateMutability": "nonpayable",
      "type": "constructor"
    },
    {
      "inputs": [],
      "name": "getFlag",
      "outputs": [
        {
          "internalType": "uint256",
          "name": "",
          "type": "uint256"
        }
      ],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        {
          "internalType": "uint256",
          "name": "toggle",
          "type": "uint256"
        }
      ],
      "name": "setFlag",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    }
  ]
mumbaiRPC = "https://rpc-mumbai.maticvigil.com"
w3 = Web3(Web3.HTTPProvider(mumbaiRPC))
pk = ''
user = w3.eth.account.from_key(pk)
dummyContract = w3.eth.contract(address=w3.to_checksum_address(contractAddress), abi=contractAbi)

def setFlag(flag):
    try:
        txn_gas = dummyContract.functions.setFlag(flag).estimate_gas({'from': user.address})
        new_transaction = {
            'nonce': w3.eth.get_transaction_count(user.address),
            'gasPrice': w3.eth.gas_price,
            'from': user.address,
            'gas': txn_gas
        }
        txn = dummyContract.functions.setFlag(flag).build_transaction(new_transaction)
        sign_txn = w3.eth.account.sign_transaction(txn, user.key)
        sent = w3.eth.send_raw_transaction(sign_txn.rawTransaction)
        w3.eth.wait_for_transaction_receipt(sent)
    except Exception as e:
        print(e)

def getFlag():
    flag = dummyContract.functions.getFlag().call({'from': user.address})
    return flag

File: scripts/dummyContract.js
const { ethers} = require("ethers");
const { StandardMerkleTree } = require("@openzeppelin/merkle-tree");
const fs = require('fs').promises;

//Network config
const contractAddress = '0x0103CfdA95f2D6084d28a935db26Ff1C5cd7DC6d'
const contractAbi = [
    {
      "inputs": [],
      "stateMutability": "nonpayable",
      "type": "constructor"
    },
    {
      "inputs": [],
      "name": "getFlag",
      "outputs": [
        {
          "internalType": "uint256",
          "name": "",
          "type": "uint256"
        }
      ],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        {
          "internalType": "uint256",
          "name": "toggle",
          "type": "uint256"
        }
      ],
      "name": "setFlag",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    }
  ]
const mumbaiRPC = "https://rpc-mumbai.maticvigil.com"
const mumbaiProvider = new ethers.JsonRpcProvider(mumbaiRPC)
const pk = ''
const mumbaiSigner = new ethers.Wallet(pk, mumbaiProvider)
const dummyContract = new ethers.Contract(contractAddress, contractAbi, mumbaiSigner)

async function setFlag( flag){
    const signer = dummyContract.connect(mumbaiSigner);
    // flag 0 sets flag to 0
    // flag 1 sets flag to 1
    // flag > 1 reverts with NVF
    const tx = await signer.setFlag(flag)
    await tx.wait()
}
async function getFlag(){
    const signer = dummyContract.connect(mumbaiSigner);
    return await signer.getFlag()
}



